### PR TITLE
Add big data streaming BBE

### PR DESCRIPTION
**Pull request by Manuri**
## Purpose
$Subject

I'm re-using @lafernando 's script at [[1]](https://wso2.com/library/knowledge-base/large-scale-data-transfer-data-services-streaming/) with a minor modification. Thanks Anjana.

### CODE REVIEW

File: examples/streaming-big-dataset/big_data_service.bal [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import ballerina/http;
+import ballerina/io;
```
Comment (anupama-pathirage) [inline examples/streaming-big-dataset/big_data_service.bal]: Remove unused import

File: examples/streaming-big-dataset/data_setup.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
+DROP DATABASE IF EXISTS StreamTestDB;
```
Comment (anupama-pathirage) [inline examples/streaming-big-dataset/data_setup.sql]: I think we cannot display this script in BBE site. @keizer619 can you please verify this?

File: examples/streaming-big-dataset/big_data_service.bal [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+// Create a HTTP service and bind to a listener on port 9090
+service dataService on new http:Listener(9090) {
+
+    resource function getData(http:Caller caller, http:Request req) {
+        http:Response res = new;
+
+        // Invoke `select` action on the mysql database endpoint and obtain a large amount of data.
+        var selectRet = testDB->select("SELECT * FROM Data", ());
+        if (selectRet is table <record {}>) {
+            // Convert the obtained data to JSON. Please note that this conversion would not load
```
Comment (anupama-pathirage) [inline examples/streaming-big-dataset/big_data_service.bal]: Shall we mention in the comment that , similarly we can convert to XML?

Comment (Manuri) [reply examples/streaming-big-dataset/big_data_service.bal]: +1. Will update

File: examples/streaming-big-dataset/big_data_service.bal [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mysql;
+
+// Create mysql client endpoint.
```
Comment (samgnaniah) [inline examples/streaming-big-dataset/big_data_service.bal]: Change to "MySQL". Apply elsewhere if applicable.

File: examples/streaming-big-dataset/big_data_service.bal (hunk 3 above)
Comment (samgnaniah) [inline examples/streaming-big-dataset/big_data_service.bal]: Change to "Note that this conversion does not load all the data into memory."

File: examples/streaming-big-dataset/big_data_service.bal [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        http:Response res = new;
+
+        // Invoke `select` action on the mysql database endpoint and obtain a large amount of data.
+        var selectRet = testDB->select("SELECT * FROM Data", ());
+        if (selectRet is table <record {}>) {
+            // Convert the obtained data to JSON. Please note that this conversion would not load
+            // all the data in to memory.
+            var jsonConversionRet = json.create(selectRet);
+            if (jsonConversionRet is json) {
+                // Set the JSON payload to the response. This will be streamed to the client once the service is
```
Comment (samgnaniah) [inline examples/streaming-big-dataset/big_data_service.bal]: Change to "This is streamed to the..". 

Avoid future tense for documentation wherever possible. Normally the word "will" indicates that the sentence is in future sense. Apply elsewhere if applicable.

File: examples/streaming-big-dataset/big_data_service.bal [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                // invoked.
+                res.setPayload(untaint jsonConversionRet);
+            } else {
+                res.setPayload({ "Error": "Error occured while converting the data to JSON" });
+                res.statusCode = http:INTERNAL_SERVER_ERROR_500;
```
Comment (samgnaniah) [inline examples/streaming-big-dataset/big_data_service.bal]: Shall we add a comment here?

File: examples/streaming-big-dataset/big_data_service.bal [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            }
+        } else {
+            res.statusCode = http:INTERNAL_SERVER_ERROR_500;
+            res.setPayload({ "Error": "Error occured while retrieving data from the database" });
+        }
+
+        var respondRet = caller->respond(res);
```
Comment (samgnaniah) [inline examples/streaming-big-dataset/big_data_service.bal]: Shall we add a comment here as well?

File: examples/streaming-big-dataset/data_setup.sql (hunk 2 above)
Comment (Manuri) [reply examples/streaming-big-dataset/data_setup.sql]: Yes, I also wanted to verify that. I suppose in that case we can point to the GitHub script location?

File: examples/streaming-big-dataset/big_data_service.description [hunk 8]
```diff
@@ -0,0 +1,12 @@
+// Ballerina inherently supports streaming large datasets. For example,
```
Comment (samgnaniah) [inline examples/streaming-big-dataset/big_data_service.description]: Ballerina inherently supports streaming large datasets. For example, retrieving a large amount of data, converting it to a preferred data format (e.g., JSON, XML) and sending it across the network on HTTP, does not require loading the entire dataset to memory at once. Data is fetched on demand from the database and incrementally converted to JSON/XML and sent to the client. This example demonstrates streaming a large amount of data through an HTTP service. Please use data_setup.sql file to setup a database with sufficient amount of data. This script creates a sample database with a table that contains an id field and two text fields, each having 1KB of data. Here, the row_count parameter of the PopulateData stored procedure is set to 100000, which would add up to around 200MB of textual data. You can increase the value of row_count parameter to increase the amount of data.

File: examples/streaming-big-dataset/big_data_service.bal [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        poolOptions: { maximumPoolSize: 5 }
+    });
+
+// Create a HTTP service and bind to a listener on port 9090
+service dataService on new http:Listener(9090) {
+
+    resource function getData(http:Caller caller, http:Request req) {
+        http:Response res = new;
+
+        // Invoke `select` action on the MySQL database endpoint and obtain a large amount of data.
```
Comment (anupama-pathirage) [inline examples/streaming-big-dataset/big_data_service.bal]: Shall we reword this to avoid action and endpoint? For ex: "Invoke `select` operation on the MySQL database client and obtain a large amount of data."
